tls/s2n_connection.h

```diff
@@ -40,6 +40,14 @@
 
 #define is_handshake_complete(conn) (APPLICATION_DATA == s2n_conn_get_current_message_type(conn))
 
+typedef enum {
+    S2N_NO_TICKET = 0,
+    S2N_EXPECTING_NEW_TICKET,
+    S2N_ATTEMPT_DECRYPT_TICKET,
+    S2N_RENEW_TICKET,
+    S2N_RECEIVED_VALID_TICKET
```

Comment:
When is S2N_RECEIVED_VALID_TICKET used? I could only find one spot where the status is set to this

Reply:
Ah sorry to make you explain further, but I'm still not sure if I'm understanding correctly. I see it in the s2n_decrypt_session_ticket function, but I still wasn't able to find it referenced in the code paths of s2n_server_extensions_send. 

I did a quick grep to see if I was missing anything and found these two references:

```
$ grep -r -C 2 S2N_RECEIVED_VALID_TICKET ./*
./tls/s2n_connection.h-    S2N_ATTEMPT_DECRYPT_TICKET,
./tls/s2n_connection.h-    S2N_NEW_TICKET,
./tls/s2n_connection.h:    S2N_RECEIVED_VALID_TICKET
./tls/s2n_connection.h-} s2n_session_ticket_status;
./tls/s2n_connection.h-
--
--
./tls/s2n_resume.c-    }
./tls/s2n_resume.c-
./tls/s2n_resume.c:    conn->session_ticket_status = S2N_RECEIVED_VALID_TICKET;
./tls/s2n_resume.c-
./tls/s2n_resume.c-    return 0;
```

Is there a place where we read the ticket status and check specifically for S2N_RECEIVED_VALID_TICKET? If not, is this state needed?

Reply:
You are right. I have removed S2N_RECEIVED_VALID_TICKET status.